rotkehlchen/chain/evm/decoding/decoder.py

```diff
@@ -111,29 +133,41 @@ def __init__(
             is_non_conformant_erc721_fn=self._is_non_conformant_erc721,
             address_is_exchange_fn=self._address_is_exchange,
         )
-        self.event_rules = [
-            self._maybe_decode_erc20_approve,
-            self._maybe_decode_erc20_721_transfer,
-        ]
-        self.event_rules.extend(event_rules)
-        self.token_enricher_rules: list[Callable] = []  # enrichers to run for token transfers
+        self.rules = DecodingRules(
+            address_mappings={},
+            event_rules=[
+                self._maybe_decode_erc20_approve,
+                self._maybe_decode_erc20_721_transfer,
+            ],
+            token_enricher_rules=[],
+            post_decoding_rules=[],
+            all_counterparties=set(self.misc_counterparties),
+        )
+        self.rules.event_rules.extend(event_rules)
         self.value_asset = value_asset
-        self.initialize_all_decoders()
+        self.decoders: dict[str, 'DecoderInterface'] = {}
+        # Recursively check all submodules to get all decoder address mappings and rules
+        rules = self._recursively_initialize_decoders(self.chain_modules_root)
+        # Sort post decoding rules by priority (which is the first element of the tuple)
+        self.rules += rules
+        self.rules.post_decoding_rules.sort(key=lambda x: x[0], reverse=True)
```

Comment:
nitpick: This should have gone above the actual sorting
```suggestion
        self.rules += rules
        # Sort post decoding rules by priority (which is the first element of the tuple)
        self.rules.post_decoding_rules.sort(key=lambda x: x[0], reverse=True)
```